website/pages/docs/k8s/installation/multi-cluster/vms-and-kubernetes.mdx

```diff
@@ -0,0 +1,293 @@
+
+If your primary datacenter is running on Kubernetes, use the Helm config from the
+[Primary Datacenter](/docs/k8s/installation/multi-cluster/kubernetes#primary-datacenter) section to install Consul.
+
+Once installed, you'll need to export the following information from the primary Kubernetes
+cluster:
+
+1. The certificate authority cert:
+
+```sh
+kubectl get secrets/consul-ca-cert --template='{{index .data "tls.crt" }}' |
+  base64 -D > consul-agent-ca.pem
+```
+
+and the certificate authority signing key:
+
+```sh
+kubectl get secrets/consul-ca-key --template='{{index .data "tls.key" }}' |
+   base64 -D > consul-agent-ca-key.pem
+```
+
+With the `consul-agent-ca.pem` and `consul-agent-ca-key.pem` files you can
+create certificates for your servers and clients running on VMs that share the
+same certificate authority as your Kubernetes servers.
+
+You can use the `consul tls` commands to generate those certificates:
+
+```sh
+$ consul tls cert create -server -dc=vm-dc
+==> WARNING: Server Certificates grants authority to become a
+    server and access all state in the cluster including root keys
+    and all ACL tokens. Do not distribute them to production hosts
+    that are not server nodes. Store them as securely as CA keys.
+==> Using consul-agent-ca.pem and consul-agent-ca-key.pem
+==> Saved vm-dc-server-consul-0.pem
+==> Saved vm-dc-server-consul-0-key.pem
+```
+
+See the help for output of `consul tls cert create -h` to see more options
+for generating server certificates.
+
+These certificates can be used in your server config file:
+
+```hcl
+# server.hcl
+cert_file = "vm-dc-server-consul-0.pem"
+key_file = "vm-dc-server-consul-0-key.pem"
+ca_file = "consul-agent-ca.pem"
+```
+
+For clients, you can generate TLS certs with:
+
+```sh
+$ consul tls cert create -client
+==> Using consul-agent-ca.pem and consul-agent-ca-key.pem
+==> Saved dc1-client-consul-0.pem
+==> Saved dc1-client-consul-0-key.pem
+```
+
+Or use the [auto_encrypt](/docs/agent/options.html#auto_encrypt) feature.
+
+1. The WAN addresses of the mesh gateways:
+
+```sh
+$ kubectl exec statefulset/consul-server -- sh -c \
+  'curl -sk https://localhost:8501/v1/catalog/service/mesh-gateway | jq ".[].ServiceTaggedAddresses.wan"'
+{
+  "Address": "1.2.3.4",
+  "Port": 443
+}
+{
+  "Address": "1.2.3.4",
+  "Port": 443
+}
+```
+
+In this example, the addresses are the same because both mesh gateway pods are
+fronted by the same Kubernetes load balancer.
+
+These addresses will be used in the server config for the `primary_gateways`
+setting:
+
+```hcl
+primary_gateways = ["1.2.3.4:443"]
+```
+
+1. If ACLs are enabled, you'll also need the replication ACL token:
+
+```sh
+$ kubectl get secrets/consul-acl-replication-acl-token --template='{{.data.token}}'
+e7924dd1-dc3f-f644-da54-81a73ba0a178
+```
+
+This token will be used in the server config for the replication token and
+can also optionally be used as the agent token since it has enough permissions.
```

Comment:
I'd suggest we call out that using this as the agent token ONLY on servers since it effectively has "root" permissions so will work but should not be used as a client agent token throughout the cluster!